components/x-live-blog-wrapper/src/LiveEventListener.js

```diff
@@ -0,0 +1,98 @@
+// TODO: parsing posts will not be needed when we use v2 route on next-live-events-api
+const parsePost = (event) => {
+	const post = JSON.parse(event.data);
+
+	if (!post || !post.mid || !post.textrendered) {
+		return;
+	}
+
+	const textElement = document.createElement('div');
+	textElement.innerHTML = post.textrendered;
+
+	const titleElement = textElement.querySelector('p strong');
+	const breakingNewsElement = textElement.querySelector('img[src*="breaking_news.gif"]');
+	const isBreakingNews = Boolean(breakingNewsElement);
+
+	if (isBreakingNews) {
+		textElement.removeChild(breakingNewsElement.parentNode);
+	}
+
+	textElement.removeChild(titleElement.parentNode);
+
+	const content = textElement.innerHTML.trim();
+	const title = titleElement.textContent;
+	const publishedTimestamp = (new Date(Number(post.emb) * 1000)).toISOString();
+
+	return {
+		postId: post.mid,
+		author: post.authordisplayname,
+		publishedTimestamp,
+		isBreakingNews,
+		isKeyEvent: Boolean(post.keytext),
+		title,
+		content
+	};
+};
+
+const dispatchLiveUpdateEvent = (eventType, data) => {
+	// consumer app will need to consume this event after the component is rendered. therefore,
+	// we defer dispatching of this event.
+	window.setTimeout(() => document.dispatchEvent(new CustomEvent(eventType, { detail: data })), 0);
```

Comment:
please don't dispatch global events from x-dash components. if this component is used in `ft-app`, they way it renders article pages means up to 3 instances of this component could be on the page at once, because of preloaded swipe panes. it looks like you should be able to use `x-interaction` for this use case though?

Reply:
The purpose of this event is to notify the consumer app when the state of this component is updated. Would that be possible with `x-interaction`?

Reply:
To be more specific;

We are listening to server sent events in this component and add new posts (by adding `x-live-blog-post` components to the markup) based on those events. 

`x-live-blog-post` component has Origami components that need to be initialised after rendering the markup. Because we want the consumer app to initialise Origami components, we notify the app using that event when the markup is changed.

Reply:
ah, okay. you could dispatch the events from the wrapper element and let them bubble instead?

Reply:
Thanks @apaleslimghost 

Sorry I forgot to update this thread; we have already updated it to raise the event from the wrapper element: https://github.com/Financial-Times/x-dash/blob/313afe7a016b2911dfabafd6bac251d104832296/components/x-live-blog-wrapper/src/LiveEventListener.js#L52

Do you think it looks ok now?

Reply:
ah! sorry, i didn't check the current state of this file. yes, that looks good!